sunpy/map/mapbase.py

```diff
@@ -1654,22 +1655,30 @@ def peek(self, draw_limb=False, draw_grid=False,
         figure.show()
 
     @toggle_pylab
-    def plot(self, annotate=True, axes=None, title=True, **imshow_kwargs):
+    @u.quantity_input(clip_interval=u.percent)
+    def plot(self, annotate=True, axes=None, title=True, clip_interval=None, **imshow_kwargs):
         """
         Plots the map object using matplotlib, in a method equivalent
         to plt.imshow() using nearest neighbour interpolation.
 
         Parameters
         ----------
-        annotate : bool
-            If True, the data is plotted at it's natural scale; with
+        annotate : `bool`, optional
+            If `True`, the data is plotted at its natural scale; with
             title and axis labels.
 
         axes: `~matplotlib.axes` or None
             If provided the image will be plotted on the given axes. Else the
             current matplotlib axes will be used.
 
-        **imshow_kwargs  : dict
+        title : `bool`, optional
+            If `True`, include the title.
+
+        clip_interval : two-element `~astropy.units.Quantity`, optional
```

Comment:
Does this *really* need to be as astropy quantity? If so and the quantity is known should probably check for it raise and error if doesn't match?

Reply:
The `u.quantity_input` decorator takes care of (some of) the error checking.

It may be slightly cumbersome for this to be a `Quantity`, but it's our guidelines to use units as much as possible.  As a case in point, for this PR I played around with some of the underlying Matplotlib machinery, and I was initially confused because I didn't know whether I was supposed to be supplying percentages or fractions as inputs, so I had to empirically figure it out.

Reply:
We are generally pretty stringent with everything being quantities where that makes sense. the `@u.quantity_input` will validate that the unit is `u.percent`

Reply:
Thumbs up on using units here though the `percent` quantity is not so useful beyond just identifying itself as such.

Reply:
Note that one could choose to specify `clip_interval=(0.01, 0.9995)*u.one` in the example in the first post and get the same clipping interval (1% to 99.95%)

Reply:
Ok, I just really, really dislike the extra typing in cases like this.

Reply:
heh:
```python
>>> (0.99*u.one).to(u.percent)                                                                                                                                                                                                 
<Quantity 99. %>
```
Didn't know that worked...